### PR TITLE
Add a log line to help track node eligibility

### LINKED ISSUE #8965: Metric label `node_scheduling_availability` does not update with changes.
### Nomad version

Nomad v0.12.5

### Issue

The `node_scheduling_eligibility` label on the Nomad client metrics does not update as the eligibility status is changed.

### Reproduction steps

Create a telemetry.nomad file with the basic configuration

```bash
telemetry {
  publish_allocation_metrics = true
  publish_node_metrics       = true
}
```

Run a dev agent with the telemetry configuration

```bash
nomad agent -dev -config=telemetry.nomad
```

Watch a metric with the label

```bash
watch "curl -s http://localhost:4646/v1/metrics | jq '.Gauges[] | select(.Name==\"nomad.client.uptime\")'"
```

Note that it appears enabled.  Set the eligibility to disabled

```bash
nomad node eligibility -self -disable
```

Watch for a stat update, observe that enabled doesn't change.

### PR DESCRIPTION
**Pull request by Kellel**
Often when debugging issues with our nomad cluster we find that it can
be difficult to correlate manual and automated actions against the
cluster state as nodes are brought into and out of production.

This doesn't fix #8965, but it is something to help minimize the inconvenience in the meantime.

### CODE REVIEW

File: nomad/node_endpoint.go [hunk 1]
```diff
@@ -661,6 +661,7 @@ func (n *Node) UpdateDrain(args *structs.NodeUpdateDrainRequest,
 	// If the node is transitioning to be eligible, create Node evaluations
 	// because there may be a System job registered that should be evaluated.
 	if node.SchedulingEligibility == structs.NodeSchedulingIneligible && args.MarkEligible && args.DrainStrategy == nil {
+		n.logger.With("node_id", node.ID).Info("Node transitioning to eligible state")
```
Comment (jrasell) [inline nomad/node_endpoint.go:664]:
```suggestion
		n.logger.Info("node transitioning to eligible state", "node_id", node.ID)
```

File: nomad/node_endpoint.go [hunk 2]
```diff
@@ -736,8 +737,10 @@ func (n *Node) UpdateEligibility(args *structs.NodeUpdateEligibilityRequest,
 	if node.SchedulingEligibility == args.Eligibility {
 		return nil // Nothing to do
 	} else if args.Eligibility == structs.NodeSchedulingEligible {
+		n.logger.With("node_id", node.ID).Info("Node transitioning to eligible state")
```
Comment (jrasell) [inline nomad/node_endpoint.go:740]:
```suggestion
		n.logger.Info("node transitioning to eligible state", "node_id", node.ID)
```

File: nomad/node_endpoint.go [hunk 3]
```diff
@@ -736,8 +737,10 @@ func (n *Node) UpdateEligibility(args *structs.NodeUpdateEligibilityRequest,
...
 		args.NodeEvent.SetMessage(NodeEligibilityEventEligible)
 	} else {
+		n.logger.With("node_id", node.ID).Info("Node transitioning to ineligible state")
```
Comment (jrasell) [inline nomad/node_endpoint.go:743]:
```suggestion
		n.logger.Info("node transitioning to ineligible state", "node_id", node.ID)
```

**Review by jrasell — CHANGES_REQUESTED** [review]
Hi @Kellel and thanks for raising this PR which seems very useful to cluster operators. I have suggested some changes for you to look over; the most notable change is to remove the use of `.With` from the logger. This function creates a sublogger with the given kv context always present which we do not need. We can just rely on the message kv formatting for this use case.

**Comment by Kellel** [comment]
Anything else needed here @jrasell ?

**Review by jrasell — APPROVED** [review]
Hi @Kellel thanks for this; I just added a changelog entry and will get this merged.

**Pull request merged**